src/client/InteractionClient.js

```diff
@@ -0,0 +1,207 @@
+ *   token: ABC,
+ *   publicKey: XYZ,
+ * });
+ *
+ * client.on('interactionCreate', () => {
+ *   // automatically handles long responses
+ *   if (will take a long time) {
+ *     doSomethingLong.then((d) => {
+ *       interaction.reply({
+ *         content: 'wow that took long',
+ *       });
+ *     });
+ *   } else {
+ *     interaction.reply('hi!');
+ *   }
+ * });
+ * ```
+ */
+class InteractionClient extends BaseClient {
+  /**
+   * @param {Options} options Options for the client.
+   * @param {undefined} client For internal use.
+   */
+  constructor(options, client) {
+    super(options);
+
+    Object.defineProperty(this, 'token', {
+      value: options.token,
+      writable: true,
+    });
+
+    Object.defineProperty(this, 'clientID', {
+      value: options.clientID,
+      writable: true,
+    });
+
+    Object.defineProperty(this, 'publicKey', {
+      value: options.publicKey ? Buffer.from(options.publicKey, 'hex') : undefined,
+      writable: true,
+    });
+
+    // Compat for direct usage
+    this.client = client || this;
+    this.interactionClient = this;
+  }
+
+  /**
+   * Get registered slash commands.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {Command[]}
+   */
+  async getCommands(guildID) {
+    let path = this.client.api.applications('@me');
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const commands = await path.commands.get();
+    return commands.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Create a command.
+   * @param {Object} command The command description.
+   * @param {Snowflake?} guildID Optional guild ID.
+   * @returns {ApplicationCommand} The created command.
+   */
+  createCommand(command, guildID) {
+    let path = this.client.api.applications('@me');
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const c = path.commands.post({
+      data: {
+        name: command.name,
+        description: command.description,
+        options: command.options.map(function m(o) {
+          return {
+            type: ApplicationCommandOptionType[o.type],
+            name: o.name,
+            description: o.description,
+            default: o.default,
+            required: o.required,
+            choices: o.choices,
+            options: o.options ? o.options.map(m) : undefined,
+          };
+        }),
+      },
+    });
+    return new ApplicationCommand(this, c, guildID);
+  }
+
+  handle(data) {
+    switch (data.type) {
+      case InteractionType.PING:
+        return {
+          type: InteractionResponseType.PONG,
+        };
+      case InteractionType.APPLICATION_COMMAND: {
+        let timedOut = false;
+        let resolve;
+        const directPromise = new Promise(r => {
+          resolve = r;
+          this.client.setTimeout(() => {
+            timedOut = true;
+            r({
+              type: InteractionResponseType.ACKNOWLEDGE_WITH_SOURCE,
+            });
+          }, 250);
```

Comment:
```suggestion
          }, 2500);
```
We have 3 seconds to send the initial respone.